### ISSUE TITLE
AAE not training the discriminator

### ISSUE THREAD

**Issue by plaut** [eriklindernoren/Keras-GAN#35]
The adversarial autoencoder doesn't seem to be training the discriminator. The discriminator accuracy remains stuck at around 50%. In addition, fake images that I generated from random Gaussian codes did not look anything like MNIST digits. In fact, they did not look any different from fake images when setting the discriminator's loss weight to zero in the autoencoder model, which is equivalent to a simple non-adversarial autoencoder.

The problem seems to be related to the batch normalization layers in the discriminator: after removing both of them, the discriminator accuracy fluctuates during training and fake images generated from random Gaussian codes look much more like real digits (I also increased the discriminator's loss weight in the autoencoder model).

**Comment by plaut** [comment]
Changing the order of the layers so that BatchNormalization is before each LeakyReLU instead of after it also seems to resolve the problem.

**Comment by 2017develper** [comment]
Keras-GAN/aae/adversarial_autoencoder.py
i want to save the model how i can do it

**Comment by eriklindernoren** [comment]
Hi,

This should be fixed in ce7dc22. Reducing the dimensionlity of the latent code made a big difference in the visual quality of generated samples. Removing the BN layers also helped. 

Cheers,
Erik
